modin/experimental/xgboost/xgboost.py

```diff
@@ -43,16 +44,47 @@ class DMatrix(xgb.DMatrix):
     Currently DMatrix supports only `data` and `label` parameters.
     """
 
-    def __init__(self, data, label):
+    def __init__(
+        self,
+        data,
+        label=None,
+        silent=False,
+        feature_names=None,
+        feature_types=None,
+        nthread=None,
+        label_lower_bound=None,
+        label_upper_bound=None,
+        feature_weights=None,
+        enable_categorical=None,
+    ):
```

Comment:
After review of xgb.DMatrix/DaskDMatrix I see that the simplest solution for the 1st iteration will be a support of parameters, which could be transmitted to the execution layer (`ModinXGBoostActor` and `_map_predict`) directly.

The list of this parameters are (this is my suggestion for this PR):
1. `feature_names`
2. `feature_types`
3. `feature_weights`
4. `enable_categorical`
5. `missing`
6. `silent`

Parameters `label_lower_bound`, `label_upper_bound`, `weights` and some other should be implemented as partitioned Modin structures and could be implemented in the separate PR.

Parameter `nthread` doesn't make sense, I think, because `ModinXGBoostActor` should use own property `_nthreads` to give better perf for internal native xgb.DMatrix.

Reply:
Now supported this parameters